packages/vc-status-list-issuer-rest-api/src/api-functions.ts

```diff
@@ -187,13 +275,9 @@ export function updateStatusEndpoint(router: Router, context: IRequiredContext,
           errorOnNotFound: true,
         })
       } else {
-        if (!updateRequest.statusListIndex || updateRequest.statusListIndex < 0) {
-          return sendErrorResponse(response, 400, 'Invalid statusListIndex supplied')
-        }
         statusListEntry = await driver.getStatusListEntryByIndex({
-          statusListIndex: updateRequest.statusListIndex,
-          ...(statusListId && { statusListId }),
-          ...(!statusListId && { correlationId: statusListCorrelationId }), // Do not query for correlationId when we have statusListId
+          statusListId: details.id,
+          ...(entryCorrelationId ? { correlationId: entryCorrelationId } : { statusListIndex: updateRequest.statusListIndex }),
```

Comment:
Why is this passing an entryCorrelationId to a statusList correlation id?

Reply:
[PERMALINK](https://github.com/Sphereon-Opensource/ssi-sdk/blob/ba7b1c04da999658e9172c86331bc0d0da784562/packages/data-store/src/types/statusList/statusList.ts#L35-L47)
There is only one correlationId so I assumed the correlationId is for the Entry and not for the list.

Shall I create two fields for this in the DB entity? statusListCorrelationId and entryCorrelationId  ?

Reply:
Yes, see other remark. The above certainly is wrong

Reply:
Ok, so we do need two correlation id fields in StatusListEntryEntity. Will have to rename both columns then.

Reply:
fixed